src/screens/SignUp.js

```diff
@@ -0,0 +1,120 @@
+import React, { Component } from "react";
+import { NavLink, } from "react-router-dom";
+import { validateEmail, validateName, validatePassword } from "../assets/helpers/userValidationTools"
+
+
+class SignUp extends Component {
+   constructor(props) {
+      super(props);
+
+      this.state = {
+         firstName: '',
+         lastName: '',
+         email: '',
+         password: '',
+         passwordVerification: '',
+         formIsValid: true,
+         redirectToHome: false,
+
+      }
+
+      this.baseState = this.state
+      this.handleValidation = this.handleValidation.bind(this)
+      this.errorSpan = this.errorSpan.bind(this)
+
+   }
+
+   userSubmit(e) {
+      e.preventDefault();
+      console.log(this.state.formIsValid)
+      if (this.handleValidation()) {
+         let { firstName, lastName, email, password, } = this.state
+         console.log(this.state)
+         fetch('http://localhost:8080/users', {
+            method: 'POST',
+            headers: {
+               Accept: 'application/json',
+               'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+               firstName: firstName,
+               lastName: lastName,
+               email: email,
+               password: password,
+               photoUrl: ""
+            }),
+         }).then((response) => response.json())
+            .then((responseJson) => {
+               console.log(responseJson);
+            })
+            .catch((error) => {
+               console.error(error);
+            })
+         alert('User Created')
+         this.resetFields()
+         console.log(this.state + "reset state")
+         this.props.history.push('/')
+
+
+
+      } else {
+         this.resetFields()
+         this.state.formIsValid.setState(true);
```

Comment:
These are also going to break the app. See above comment

Reply:
this doesn't break the app.

Reply:
you were absolutely right, I didn't try it with a totally empty form. My bad.